### PR TITLE
Fix: SPL Head command followed by Stats When querying less than total Segments

### PR DESCRIPTION
**Pull request by Macbeth98**
# Description
- The SPL Head command followed by stats was not working when we query or are given a limit less than the total segments.
- For example, the below queries will not work:
```SPL
* | head 250000 | stats count(*)
* | head 20000 | stats sum(latency)
* | head 30000 | stats sum(latency) by http_method
```
- The above queries won't work assuming that there are more segments than required because of the limit.

#### Fixed a bug in the Group By Recs Aggregation function: `PerformGroupByRequestAggsOnRecs` in `searchaggs.go`. 
- Before, this function only returned the aggregations that were only part of the final segment.
- Now, this is fixed. It now stores the previous aggregations in `NodeResult` in struct and appends data to it for each segment.

# Testing
- Tested the code against the UI.
- Written Tests for Stats and Stats With Group By with Size Limit in `searchaggs_test.go`

# Checklist:

- [x] I have self-reviewed this PR.
- [x] I have removed all print-debugging and commented-out code that should not be merged.
- [x] I have added sufficient comments in my code, particularly in hard-to-understand areas.
- [x] I have formatted the code, if applicable. For Go, I have run `goimports -w .`.

### CODE REVIEW

File: pkg/segment/search/searchaggs_test.go [hunk 1]
```diff
@@ -0,0 +1,363 @@
...
+		resultMap := PerformAggsOnRecs(nodeResult, aggs, recs, finalCols, uint64(numSegements), true, 1)
+
+		assert.Equal(t, uint64(i+1), nodeResult.RecsAggsProcessedSegments, "expected=%v, actual=%v", i+1, nodeResult.RecsAggsProcessedSegments)
+
+		if i == numSegements-1 {
+			assert.Equal(t, 1, len(resultMap), "expected=%v, actual=%v", 1, len(resultMap))
+			assert.True(t, resultMap["CHECK_NEXT_AGG"], "expected=%v, actual=%v", true, resultMap["CHECK_NEXT_AGG"])
+			assert.Equal(t, recsSize, len(recs), "expected=%v, actual=%v", recsSize, len(recs))
+			for _, record := range recs {
+				assert.Equal(t, uint64(2), record["count(*)"], "expected=%v, actual=%v", recsSize*numSegements, record["count(*)"])
```
Comment (AndrewHess) [inline pkg/segment/search/searchaggs_test.go:216]: The two expected values (`uint64(2)` and `recsSize*numSegements`) are different

File: pkg/segment/search/searchaggs_test.go [hunk 2]
```diff
@@ -0,0 +1,363 @@
...
+	finalCols["measure2"] = true
+	finalCols["measure3"] = true
+
+	return recs, finalCols
+}
+
+// Tests the case where the size limit is zero and the number of segments is 2.
+// Normal case where sizelimit should not be considered.
+func Test_PerformMeasureAggsOnRecsSizeLimit_Zero_MultiSegments(t *testing.T) {
+	numSegements := 2
```
Comment (AndrewHess) [inline pkg/segment/search/searchaggs_test.go:42]: typo: should be `segment` not `segement`. Same comment elsewhere in this PR

**Review by AndrewHess — COMMENTED** [review]
Looks good, just a few nits to fix

File: pkg/segment/search/searchaggs_test.go (hunk 1 above)
Comment (Macbeth98) [reply pkg/segment/search/searchaggs_test.go:216]: Yes, I will update.

**Review by AndrewHess — APPROVED** [review]
Looks good

**Pull request merged**